recipe/meta.yaml

```diff
@@ -26,6 +26,7 @@ requirements:
     - libssh2
     - openssl        # [unix]
     - libnghttp2     # [unix]
+    - libpsl  # [unix]
```

Comment:
Can you comment on why this is needed? Is this something we should pin globally to avoid symbols mismatch?

Reply:
https://github.com/curl/curl/pull/12661

it looks like the alternative would be to explicitly disable libpsl.

Reply:
So either this is needed or actively disable the PSL functionality, the background read on the decision by the curl maintainers is written here:
https://daniel.haxx.se/blog/2024/01/10/psl-in-curl/

Considering that PSL was never enabled on the curl build on coda-forge I can also change this PR to actively disable PSL and potentially open another PR for discussion of adding PSL (or not, depending on decision).

Reply:
The latter might be faster.  As @ocefpaf said, we may need to have a global pin for libpsl before it would be safe add it as a dependency.

Reply:
Pinging @isuruf and @h-vetinari here to get a few more data points on this but yes, the latter would get a fast merge.